README.md

```diff
@@ -20,127 +19,304 @@ let thirdNumber = numbers[2]
+- I think it's important for them to see the error they would receive IF they tried updating a value at a certain index that is out of bounds. Or if they try accessing an element that could be out of bounds. They might then wonder HOW they can in fact make sure they are within bounds and we COULD show them this but I'm very very reluctant in explaining to them exactly what is going on here as the next lesson will introduce optionals (which is usually the hardest thing for them to grasp). **OPINION WANTED**: Do you think we should at the very least show them something like this?:
 
 ```swift
  if let index = names.indexOf("Jim") {
     names.removeAtIndex(index)
  }
- 
+
  // ["Sarah", "Bran", "Sansa"]
 ```
 
-* Referring to `count`, they should be exposed to the following where it is explained thoroughly what is going on: 
+- Referring to `count`, they should be exposed to the following where it is explained thoroughly what is going on:
 
 ```swift
 let rainbowColors = ["Red", "Orange", "Yellow", "Green", "Blue", "Indigo", "Violet"]
- 
+
  let numberOfColors = rainbowColors.count
  // 7
- 
+
  print("There are \(numberOfColors) colors in the rainbow.")
  // prints "There are 7 colors in the rainbow.
- 
+
  if rainbowColors.count > 5 {
     print("There are more than 5 colors in a rainbow.")
  }
  //prints "There are more than 5 colors in a rainbow."
 ```
 
-* Referring to `isEmpty` - I like the idea of stepping them through the following two examples:
+- Referring to `isEmpty` - I like the idea of stepping them through the following two examples:
 
 ![deliLine](http://i.imgur.com/ivgaSon.png?1)
+
 ```swift
  var deliLine = [
     "Albert Einstein",
     "Isaac Newton",
     "Galileo Galilei",
     "Marie Curie"
  ]
- 
+
  if deliLine.isEmpty {
     print("The deli line is empty")
  } else {
     print("There are people waiting to eat!")
  }
- 
+
  // prints "There are people waiting to eat!"
 ```
+
 ![Moons](http://i.imgur.com/reNsnn6.png?1)
+
 ```swift
  let moonsOfJupiter = [
     "Europa",
     "Ganymedge",
     "Io",
     "Callisto"
 ]
- 
+
  if !moonsOfJupiter.isEmpty {
     for moon in moonsOfJupiter {
         print("\(moon) is a moon orbiting Jupiter.")
     }
  }
- 
+
  // Europa is a moon orbiting Jupiter.
  // Ganymede is a moon orbiting Jupiter.
  // Io is a moon orbiting Jupiter.
  // Callisto is a moon orbiting Jupiter.
 ```
 
+--------------------------------------------------------------------------------
+
+In Unit 1 you saw the following:
+
```

Comment:
@JimCampagno instead of saying what shown in unit 1, just say what this is here and explain what the function is doing. i.e. If we wanted a function that would take in a string and make it uppercase, we can write it like this: